.travis.yml

```diff
@@ -1,33 +1,66 @@
 language: cpp
 sudo: false
-compiler: clang
-#    - clang
-#    - gcc
+compiler:
+    - clang
+    - gcc
 
-os: osx
-#    - linux
-#    - osx
+os:
+    - linux
+    - osx
+#osx_image: xcode7
+dist: trusty
+
+addons:
+  apt: 
+   sources:
+      - boost-latest
+      - ubuntu-toolchain-r-test
+   packages:
+      - gcc-4.8
+      - g++-4.8
+      - libboost1.55-all-dev
+      - libilmbase-dev
+      - libopenexr-dev
+      - libgif-dev
+      - libtiff4-dev
+      - libopenjpeg-dev
+      - libwebp-dev
+      - ffmpeg
 
 cache:
     ccache: true
+    apt: true
     directories:
     - $HOME/.ccache
     - $HOME/lgritz/oiio-images
     - $HOME/lgritz/libtiffpic
 
 before_install:
-    - src/travis/before_install.bash
+    - if [ $TRAVIS_OS_NAME == osx ] ; then sysctl machdep.cpu.features ; fi
+    - if [ $TRAVIS_OS_NAME == linux ] ; then cat /proc/cpuinfo ; fi
 
 install:
-    - src/travis/install.bash
+    - if [ "$CXX" == "g++" ]; then export CXX="g++-4.8" ; fi
+    - if [ $TRAVIS_OS_NAME == osx ] ; then src/build-scripts/install_homebrew_deps.bash ; fi
+    - if [ $TRAVIS_OS_NAME == linux ] ; then src/build-scripts/travis_build_openexr.bash ; fi
+    - if [ $TRAVIS_OS_NAME == linux ] ; then export ILMBASE_HOME=${HOME}/openexr-install ; export OPENEXR_HOME=${HOME}/openexr-install ; fi
+    - src/build-scripts/install_test_images.bash
 
 # before_script:
 
-script: make VERBOSE=1 cmakesetup && make -j 2 && make test
+# Strategy: make a full default build + test (default to sse2 and C++03)
+# Then just check that the build works fine for sse4.2.
+# And finally, do a build and test with C++11.
+script:
+    - make VERBOSE=1 cmakesetup && make -j 2 && make test
+    - make nuke && make -j 2 USE_SIMD=sse4.2 && make test
+    - make nuke && make -j 2 USE_CPP11=1 && make test
```

Comment:
You're correct about the dependencies. Unfortunately, [Amdahl's Law](https://en.wikipedia.org/wiki/Amdahl's_law) springs to mind here. If the cost of dependencies far outweighs the test/build over-head, this wouldn't improve anything. 

However, if you're running `make nuke` everytime, isn't the build rebuilding all of the internal dependencies anyway? For things like OpenEXR 2.*, here is a suggestion:

```yml
env:
  - BUILD_TYPE="default"
  - BUILD_TYPE="cpp11"
  - BUILD_TYPE="simd"
```

Within the build OpenEXR script, instead of installing it into `${HOME}/openexr-install`, installing it into `.ccache/${BUILD_TYPE}/openexr-install`. When the script runs, run `make install` against the `.ccache/` directory instead of building it from scratch. The `make install` should override whichever version the container had. 

I'm just spit-balling, I'd suggest going with what you have and perhaps keeping these ideas as a future improvement! If I'm not being helpful, please let me know.